pkg/actuators/machine/utils.go

```diff
@@ -398,3 +403,97 @@ func GetMibValueOfQuantity(quantity resource.Quantity) int64 {
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+// GetGPU returns the VMGpu for the given Prism Element (uuid) and GPU identifier input
+func GetGPU(ctx context.Context, client *nutanixClientV3.Client, peUUID string, gpu machinev1.NutanixGPU) (*nutanixClientV3.VMGpu, error) {
+	peGPUs, err := GetGPUsForPE(ctx, client, peUUID)
+	if err != nil {
+		return nil, err
```

Comment:
Would appreciate if we can wrap errors at every level to add additional context. Makes it easier to debug as you can search for the various messages in the final message and find the stack

Reply:
addressed.